test/e2e_node/device_plugin.go

```diff
@@ -163,6 +164,109 @@ var _ = framework.KubeDescribe("Device Plugin [Feature:DevicePlugin] [Serial] [D
 	})
 })
 
+// Serial because the test restarts Kubelet
+// Slow because this is a soak test
+var _ = framework.KubeDescribe("Device Plugin Soak [Feature:DevicePluginSoak] [Soak] [Slow] [Serial] [Disruptive]", func() {
+	f := framework.NewDefaultFramework("device-plugin-errors")
+
+	Context("DevicePlugin", func() {
+		By("Enabling support for Device Plugin")
+		tempSetCurrentKubeletConfig(f, func(initialConfig *kubeletconfig.KubeletConfiguration) {
+			initialConfig.FeatureGates[string(features.DevicePlugins)] = true
+		})
+
+		const (
+			// how many waves of device plugins
+			scale = 10
+			// how many milliseconds to wait before dumping the next wave of device plugins
+			millisecondsBetweenWaves = 500 * time.Millisecond
+			// ping to check device plugin health periodicity and continuously
+			pingPeriodInSeconds = 15 * time.Second
+			// timeout of the whole soak test
+			soakTimeout = 30 * time.Minute
+		)
+
+		describtion := fmt.Sprintf("Kubelet should survive after handling multi device plugins for a duration of %v, "+
+			"scaling up to %v device plugins per node.", soakTimeout, scale)
+
+		It(describtion, func() {
+			defer GinkgoRecover()
+			var wg sync.WaitGroup
+			wg.Add(scale)
+			for i := 0; i < scale; i++ {
+				go func() {
+					wave := fmt.Sprintf("wave-%v", strconv.Itoa(i))
+					framework.Logf("Starting device plugin soak test, wave = %v", wave)
+					setUpAndConsumeResourceWithSleepOf(f, pingPeriodInSeconds, wave, soakTimeout)
+					framework.Logf("Completed device plugin soak test, wave = %v", i)
+					wg.Done()
+				}()
+				// gab between waves
+				time.Sleep(millisecondsBetweenWaves)
```

Comment:
why a gap here? will it fail without gap?

Reply:
It is just  a niceness here, It isn't expected to fail without gap.

Reply:
that I would suggest not having a gap here~